kubernetes-client/src/main/java/io/fabric8/kubernetes/client/utils/Serialization.java

```diff
@@ -253,17 +253,37 @@ public Type getType() {
   }
 
   private static boolean containsMultipleDocuments(String specFile) {
-    Pattern p = Pattern.compile(DOCUMENT_DELIMITER);
-    Matcher m = p.matcher(specFile);
-    int count = 0;
-    while (m.find()) {
-      count++;
+    String[] documents = splitSpecFile(specFile);
+    int nValidDocuments = 0;
+    for(String document : documents) {
+      if(validate(document))
+        nValidDocuments++;
     }
-    if (count == 1) {
-      String[] documents = specFile.split(DOCUMENT_DELIMITER);
-      return validate(documents[0]);
+
+    return nValidDocuments > 1;
+  }
+
+  private static String[] splitSpecFile(String aSpecFile) {
+    List<String> documents = new ArrayList<>();
+    String[] lines = aSpecFile.split(System.lineSeparator());
+    int nLine = 0;
+    StringBuilder builder = new StringBuilder();
+
+    while(nLine < lines.length) {
+      if(lines[nLine].length() >= DOCUMENT_DELIMITER.length()
```

Comment:
@piyush1594, doesn't this split the `aSpecFile` on every line which is shorter than the delimiter?

Especially empty lines which you might use for logical separation.

Reply:
This is separating all the new lines, adding them to an array which is `lines` and then iterating over the `lines` array.

You might be facing some issue because of this if a line's length is less then delimiter length that is already fixed by #1070

Reply:
Kubernetes Client 3.1.12 released with the fix